projects/RabbitMQ.Client/client/framing/BasicProperties.cs

```diff
@@ -166,69 +284,57 @@ public override string ClusterId
 
         public override void ClearClusterId() => _clusterId = default;
 
-        public override bool IsContentTypePresent() => _contentType != default;
+        public override bool IsContentTypePresent() => _contentType != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b1000_0000) != 0);
+
+        public override bool IsContentEncodingPresent() => _contentEncoding != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b0100_0000) != 0);
+
+        public override bool IsHeadersPresent() => _headers != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b0010_0000) != 0);
 
-        public override bool IsContentEncodingPresent() => _contentEncoding != default;
+        public override bool IsDeliveryModePresent() => _deliveryMode != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b0001_0000) != 0);
 
-        public override bool IsHeadersPresent() => _headers != default;
+        public override bool IsPriorityPresent() => _priority != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b0000_1000) != 0);
 
-        public override bool IsDeliveryModePresent() => _deliveryMode != default;
+        public override bool IsCorrelationIdPresent() => _correlationId != default || (_readOnlyMemory.Length > 0 && (_readOnlyMemory.Span[0] & 0b0000_0100) != 0);
```

Comment:
Can we somehow collapse the hierarchy and thus I line those methods?

Reply:
I was wondering about that actually, I was thinking that calculating the IsPresent and Offsets to read the properties could be loaded all at once when the first property is accessed to also prevent calculations from being repeated. It should be easy to read and calculate those without allocating on first access.

Reply:
It's possible to read all the is present values into a flags enum bypassing all the read bits related code.
For an example:

https://github.com/YulerB/rabbitmq-dotnet-client/blob/master/projects/client/RabbitMQ.Client/src/client/Protocol/BasicProperties.cs